### PR TITLE
test: add coverage for DeepLinkViewModel error states

### PR DESCRIPTION
**Pull request by nonproto**
💡 What: Added `DeepLinkViewModelTest.kt` with a unit test targeting the "sad path" in `handleDeepLink`.
🎯 Why: To improve test coverage by validating edge cases. Specifically, ensuring that missing UUID mappings correctly result in a `DeepLinkState.Error` with the appropriate error message, protecting against regressions in the app's deep link routing logic.

---
*PR created automatically by Jules for task [8571812100257383768](https://jules.google.com/task/8571812100257383768) started by @nonproto*

### CODE REVIEW

File: app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private lateinit var mockMangaMappings: MangaMappings
+    private lateinit var mockSourceManager: SourceManager
+    private lateinit var mockMangaDex: MangaDex
+    private lateinit var mockMangaRepository: MangaRepository
+
+    private lateinit var viewModel: DeepLinkViewModel
+
+    @Before
+    fun setup() {
+        Dispatchers.setMain(testDispatcher)
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt:38]: > high
> 
> Since `Injekt` is a global registry, calling `Injekt.addSingleton` without resetting it will cause `DoubleRegistrationException` when multiple tests are run in the same suite. To prevent this, re-initialize the global `Injekt` variable with a new `KotlinInjektRegistrar()` at the start of each test setup.

@jules

File: app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.nekomanga.data.database.repository.MangaRepository
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.addSingleton
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt:22]: > medium
> 
> To support resetting the global `Injekt` instance before each test, import `KotlinInjektRegistrar`.

@jules

File: .jules/testpilot.md [hunk 3]
```diff
@@ -11,3 +11,5 @@
...
 **Action:** Always mock the intermediate `Preference` object and its `.set()` or `.get()` methods individually when testing domain logic that alters App preferences to prevent runtime test crashes.
 ## 2025-02-23 - [UpdateTrackChapterTest] **Learning:** Mocking simple data classes (`mockk<TrackItem>(relaxed = true)`) in Kotlin can lead to test fragility and code smells. **Action:** Instantiate real instances of simple data classes in tests instead of mocking them.
+
+## 2024-05-18 - DeepLinkViewModel Testing DI **Learning:** When writing tests for classes in Nekomanga that use Injekt (like ViewModels), `InjektRegistrar` cannot be instantiated directly as it is an interface. Use `Injekt.addSingleton()` directly. **Action:** Next time setting up tests involving `Injekt`, just call `Injekt.addSingleton(mock)` directly instead of trying to override the `Injekt` singleton instance with `InjektRegistrar`.
```
[the remark this answers is not in the document]
Comment (nonproto) [reply .jules/testpilot.md]: > medium
> 
> Update the learning and action to reflect that `Injekt` can be reset by reassigning it to a new `KotlinInjektRegistrar()` instance, which is the standard implementation of `InjektRegistrar`.

@jules
